src/Shell.c

```diff
@@ -375,6 +430,20 @@ void MI_CALL Shell_Load(Shell_Self** self, MI_Module_Self* selfModule,
         GOTO_ERROR("out of memory", MI_RESULT_SERVER_LIMITS_EXCEEDED);
     }
 
+    /* Initialize the environment
+     *
+     * This is necessary because OMI can launch this process as a non-root user;
+     * but OMI does not do anything to ensure the environment is correct.
+     * For instance, if HOME=/root in omiserver, * this omiagent process inherits the value.
+     * However, for a shell provider, this causes problems.
+     * PowerShell depends on HOME to be correct; i.e. the user's home folder,
+     * and throws an exception if it gets an IO error on HOME.
+     *
+     * Here we lookup the user's home folder via getpwuid(),
+     * and set HOME for our process to the correct value.
+     */
+    (*self)->home = SetHomeDir();
```

Comment:
@paulcallen Also here, I'm not sure what style you prefer. I could pass `self->home` instead, and think that might be cleaner. It just ties `SetHomeDir` and self a bit more tightly.